numpy-stubs/__init__.pyi

```diff
@@ -813,14 +814,27 @@ _ScalarNumpy = Union[generic, dt.datetime, dt.timedelta]
 _ScalarBuiltin = Union[str, bytes, dt.date, dt.timedelta, bool, int, float, complex]
 _Scalar = Union[_ScalarBuiltin, _ScalarNumpy]
 
-_ScalarGeneric = TypeVar(
-    "_ScalarGeneric", bound=Union[dt.datetime, dt.timedelta, generic]
+# Integers and booleans can generally be used interchangeably
+_ScalarInt = TypeVar("_ScalarInt", bound=Union[integer, bool_])
+_ScalarGeneric = TypeVar("_ScalarGeneric", bound=generic)
+_ScalarGenericPlus = TypeVar(
```

Comment:
Similarly `_ScalarGenericPlus` is not very descriptive of the type here. Also this definition would allow passing through subclasses of `datetime` and `timedelta`; are we confident that will work correctly?

Reply:
Maybe something like ``_ScalarGenericDT`` instead?

About your second concern, some quick testing suggests that subclassing does indeed work:
``` python
>>> import datetime as dt
>>> import numpy as np

>>> class A(dt.timedelta): ...
>>> np.take(A(1), 0).__class__
__main__.A

>>> class B(dt.datetime): ...
>>> np.take(B(2000, 1, 1), 0).__class__
__main__.B
```